### PR TITLE
Add debug event support

### PR DESCRIPTION
**Pull request by cdhoffmann**
The SDKs often needs to change its standard business logic to enable debugging/testing workflows. We need an event structure that external systems can follow to generate an event that communicates to the listening SDK extension that it should operate in a debug mode - a mode outside of expected production behavior if one exists.

This commit adds an eventSource "com.adobe.eventSource.debug" to identify such events.
Further, it adds utility methods to identify the eventType and eventSource provided within the `data.debug` content of such events.
**Structure:**
```
{
        "name" : "......",
        "type" : "com.adobe.eventType.system",
        "source" : "com.adobe.eventSource.debug",
        "data" : {
            "debug": {
                "eventType": "<DEBUG_EVENT_TYPE>",
                "eventSource": "<DEBUG_EVENT_SOURCE>"
            },
             "someOtherKey":  {
               ....
               }
         }
}
```

Usage:

```
var debugType = Event.debugEventType
var debugSource = Event.debugEventSource
```

## Motivation and Context
The SDKs often needs to change its standard business logic to enable debugging/testing workflows. We need an event structure that external systems can follow to generate an event that communicates to the listening SDK extension that it should operate in a debug mode - a mode outside of expected production behavior if one exists.

This commit adds an eventSource "com.adobe.eventSource.debug" to identify such events.
Further, it adds utility methods to identify the eventType and eventSource provided within the `data.debug` content of such events.

## How Has This Been Tested?
- Unit tests

## Screenshots (if appropriate):
n/a
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by cdhoffmann** [comment]
Ran swiftFormat and fixed a bunch of whitespace issues, apologies for the extra file changes.

File: AEPCore/Sources/core/Event+Debug.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        return nil
+    }
+    
+    /// The data found in the event if the event is a debug event, otherwise nil
+    var debugEventData: [String: Any]? {
+        if type != EventType.system || source != EventSource.debug {
+            return nil
+        }
+        
+        return data
```
Comment (praveek) [inline AEPCore/Sources/core/Event+Debug.swift:46]: You should return the contents of data?["debug]

File: AEPCore/Sources/core/Event+Debug.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    var debugEventType: String? {
+        if let debugDictionary = data?[CoreConstants.Keys.DEBUG] as? [String: Any], let debugType = debugDictionary[Event.DEBUG_EVENT_TYPE_KEY] as? String {
+            return debugType
+        }
+        return nil
+    }
+    
+    /// The debug event source (identified by debug.eventSource) found in the event data if present, nil otherwise
+    var debugEventSource: String? {
+        if let debugDictionary = data?[CoreConstants.Keys.DEBUG] as? [String: Any], let debugSource = debugDictionary[Event.DEBUG_EVENT_SOURCE_KEY] as? String {
```
Comment (praveek) [inline AEPCore/Sources/core/Event+Debug.swift:34]: You can reuse debugEventData here

File: AEPCore/Sources/core/CoreConstants.swift [hunk 3]
```diff
@@ -54,6 +54,7 @@ enum CoreConstants {
         static let EXPERIENCE_CLOUD_SERVER = "experienceCloud.server"
         static let ADVERTISING_IDENTIFIER = "advertisingidentifier"
         static let PUSH_IDENTIFIER = "pushidentifier"
+        static let DEBUG = "debug"
```
Comment (praveek) [inline AEPCore/Sources/core/CoreConstants.swift:57]: Can you also keep this constant local to Event+Debug?

File: AEPCore/Sources/core/Event+Debug.swift (hunk 1 above)
Comment (cdhoffmann) [reply AEPCore/Sources/core/Event+Debug.swift:46]: Ah, thought we were making a helper for other extensions to get the event data ONLY if it was debug event type. My bad.

File: AEPCore/Sources/core/Event+Debug.swift [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+///
+public extension Event {
+    
+    static let DEBUG_EVENT_DEBUG_KEY = "debug"
+    static let DEBUG_EVENT_TYPE_KEY = "eventType"
+    static let DEBUG_EVENT_SOURCE_KEY = "eventSource"
+    
+    /// The debug event type (identified by debug.eventType) found in the event data if present, nil otherwise
+    var debugEventType: String? {
+        if let debugDictionary = debugEventData, let debugType = debugDictionary[Event.DEBUG_EVENT_TYPE_KEY] as? String {
```
Comment (praveek) [inline AEPCore/Sources/core/Event+Debug.swift:27]: nit 
```
if let debugType = debugDictionary?[Event.DEBUG_EVENT_TYPE_KEY] as? String] {
   ...
}
```

**Review by praveek — APPROVED** [review]

File: AEPCore/Sources/core/Event+Debug.swift [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    static let DEBUG_EVENT_TYPE_KEY = "eventType"
+    static let DEBUG_EVENT_SOURCE_KEY = "eventSource"
+    
+    /// The debug event type (identified by debug.eventType) found in the event data if present, nil otherwise
+    var debugEventType: String? {
+        if let debugType = debugEventData?[Event.DEBUG_EVENT_TYPE_KEY] as? String {
+            return debugType
+        }
+        return nil
+    }
```
Comment (sbenedicadb) [inline AEPCore/Sources/core/Event+Debug.swift:31]: i think we can just clean these up like this: 
```
    var debugEventType: String? {
        debugEventData?[Event.DEBUG_EVENT_TYPE_KEY] as? String
    }
```

**Review by sbenedicadb — APPROVED** [review]
looks good with one little recommendation to clean up the implementation of the extension vars

**Review by sbenedicadb — APPROVED** [review]

**Pull request merged**